### ISSUE TITLE
Inconsistent Inspector panel behavior when clicking a node vs. a file

### ISSUE THREAD

**Issue by BigZaphod** [godotengine/godot-proposals#6460]
### Describe the project you are working on

I'm just working on hobby projects of no consequence.

### Describe the problem or limitation you are having in your project

I am constantly getting momentarily confused about what's appearing in the Inspector tab. If I single click a node in the scene tree, that node becomes selected and the information for that node is in the Inspector tab as I would expect. HOWEVER if I single click a file in the FileSystem tab, all that happens is that the file becomes visually selected - it does not change the content of the Inspector tab despite my having just clicked it and despite there being a blue outline around the entire FileSystem tab's inner content indicating that focus is there.

This screenshot isn't terribly exciting, but it shows what happens *after* I clicked on the "icon.svg" file in the FileSystem tab. Note that the Inspector is still showing the information for a node I had selected previously. Also note that a quick glance at this window makes it appear as if two different objects are selected at the same time - the CanvasGroup node in the scene, and also the icon.svg file.

<img width="1338" alt="Screenshot 2023-03-08 at 2 59 56 PM" src="https://user-images.githubusercontent.com/453721/223849773-78a135bc-0d05-472e-833d-1e477a719d96.png">

This is very confusing, IMO.

There are multiple problems here, I think:

One is that single clicking on a file in the FileSystem tab only selects it in that tab but doesn't open the information for that file in the Inspector. I instead have to double click file objects to show them in the Inspector tab - unlike everything else which is a single click (as far as I know).

The other problem is that multiple things can look selected at the same time even when they are not. In the screenshot above, the CanvasGroup node shouldn't look selected because I had just clicked on the icon.svg file and that is where my focus is. You could argue that this is fine because I didn't "fully select" icon.svg by double clicking it - but indeed even when I do double click icon.svg to "open" the file, now it shows up only in the inspector but nothing else has changed and no new editor has appeared - it still looks like CanvasGroup is selected as well except now the Inspector properties don't make sense. And even worse, because my mouse happened to enter the main editor zone of the window, there's now a thin blue outline around it making me think the editor context is active!

<img width="1338" alt="Screenshot 2023-03-08 at 3 13 14 PM" src="https://user-images.githubusercontent.com/453721/223851351-1f14a7d3-81c9-415a-ad3c-9753665b7c53.png">

I realize that there's probably some complication here because double clicking a file is how you "open" it and I don't think I would want a single click of a scene file to immediately open it up in a new tab (although... maybe that wouldn't be terrible?), but this is fuzzy when it comes to other types of files like images and resources which appear to "open" in the inspector.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I don't entirely know what the solution should be, just that the current situation can be confusing and disorienting. For reference, I have more experience with Unity and in Unity when I select something the previously selected thing generally becomes unselected even if I'm selecting a different thing in a different tab/panel. In my experience, that's pretty standard behavior for most apps and yet Godot presents the selections as if each individual tab/panel is its own little self-contained world unaware of the environment it exists within. (Mostly that's probably good - but this is one case where it isn't, IMO.)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I don't really have a specific proposal, just an observation/pain-point. (Note that I originally submitted this as a bug/issue https://github.com/godotengine/godot/issues/74620 and my impression from there was that I should submit it here instead despite the fact that the form *here* is asking for things like a detailed design of my proposal when all I'm trying to do is get a usability difficulty in front of the people who might actually be able to do something about it without having to argue!)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

The workaround is to just get used to how it is now, but I don't think that's the best default way to think about this sort of thing.

### Is there a reason why this should be core and not an add-on in the asset library?

I assume this would have to be addressed in the core? I have no idea. As I said above, I'm just trying to point out a usability difficulty I encountered but was shooed out of submitting it as an issue on the main repo and sent here instead. Honestly it kind of feels like the project doesn't even want this kind of feedback so maybe this is futile.

**Comment by Calinou** [comment]
How is drag-and-drop supposed to work if single-clicking on a resource in the FileSystem dock switches the inspector view?

**Comment by BigZaphod** [comment]
> How is drag-and-drop supposed to work if single-clicking on a resource in the FileSystem dock switches the inspector view? You can drag and drop resources from the FileSystem dock to relevant Inspector properties after all 🙂
> 
> Remember that the Godot editor can run on mobile platforms too (with touch controls). We should avoid requiring users to select another file, then click on another file without releasing the mouse button while dragging the file to the resource in the Inspector.

I'm aware of this. I'm a mobile developer and have done iOS development since before there was a public SDK and the tension between clicking and dragging is very familiar to me and a solved thing. There's no difficultly here that I can see except for the one Godot seems to make for itself: Godot's UI appears to change selection on touch/mouse-down rather than touch/mouse-up. That's it. This part is not a real problem, IMO - it just seems like one because of Godot's UI convention which I may argue is misguided.

A drag is not a tap or a click. A tap and click should not be recognized until touch/mouse is *released* and something like changing the currently selection should only happen on a tap/click. This means when you start dragging beyond a threshold (a few pixels) the action becomes a drag instead and no click action will be generated even if the drag is cancelled or dropped.

**Comment by timothyqiu** [comment]
> Godot's UI appears to change selection on touch/mouse-down rather than touch/mouse-up.

Yeah, select on mouse up is how we handled drag-and-drop of the nodes from the Scene dock. So I don't think drag-and-drop would be an issue if we decide to adapt the click-to-show-in-inspector behavior.

> I don't think I would want a single click of a scene file to immediately open it up in a new tab

You could also try opening a font file or glTF/FBX file. Once selected in FileSystem, you are allowed to change the import options in the Import dock. If you double click them, you will see a separate import configure dialog. This means implement this proposal would need a redesign of the current workflow.

**Comment by pie75** [comment]
I personally think the most impactful part of this is that the difference between 'selected' and 'active' isn't clearly shown.

I would suggest changing Godot's UI philosophy so that:
'active' is **outlined** with clear background
'selected' keeps the highlighted background.

This should help in other cases, too, if they arise, without making it less readable what is active or not.

**Comment by xiaden** [comment]
I am a first-time user of Godot.  The very first thing I noticed that annoyed me was this behavior.  It's problematic to me, but I can also see how any implemented fix that actually changes selection behavior would be problematic for virtually anyone else.  

The first place I went to when trying to get the editor to work in my desired workflow was the options menu, looking for a check box to "inspect files on single click".  The first place I looked was Interface, Editor. Second place was Interface, Inspector.  Then it was basically a 10-minute bout of "Where would this actually exist in this options menu", followed by googling how to change the behavior.  Which lead me here, to log into a very old Github account and start this rant.

As such, my proposal is to add a check box to the Editor tab (or inspector tab, it honestly sounds more like it belongs there) in the Interface group of the options menu that would toggle the behavior, off being a double click to inspect files, on being a single click, and defaulting to off.  Keeps the interface in line with current user expectations, and allows us oddballs who want inspector to work differently to have our cake.

**Comment by dougVanny** [comment]
This has been a big struggle for me. I'm used to clicking just once and already going to the inspector to make my changes. This is specially bothersome when working with multiple resources of the same type, where I believe I am changing multiple files but instead keep changing just one I first opened over and over

**Comment by passivestar** [comment]
> A drag is not a tap or a click. A tap and click should not be recognized until touch/mouse is released

^This

This inconsistency in behaviour between the Scene tab and the FileSystem tab was the first thing that I noticed when I started using Godot. You can drag nodes from the Scene tab onto inspector with no problem because it doesn't switch unless you let go. I don't see any reason for the FileSystem tab to behave differently, it could just do the same thing that Scene does

**Comment by synthc** [comment]
This is actually quite impactful, because double clicking a script also opens/focuses that script in your code editor.  This overloaded behavior is going to be a blocker for Unity developers like me who are looking to switch to Godot and want to be able to implement their custom inspector interfaces.

As has already been suggested, there should be an option to single-click inspect files.
